### DISCUSSION TITLE
what's the correct way to put a collection in a select?

### DISCUSSION THREAD

**Discussion by KevinEscobarV** [wireui/wireui#297] [Q&A] [answered]
**If I save the query in a variable, what is the correct way to display this data in a select?**

I know I can alternatively use a select with slot options, but i would like to use this way

```
 public function render()
    {
        $methods = PaymentMethod::all();
        return view('livewire.payment', compact('methods'));
    }
```
**I have tried this way but it doesn't work**
```
<x-select
    label="Select a payment method"
    placeholder="Select an option"
    :options="{{ $methods }}"
    wire:model.defer="payment_method"
/>
```
thanks for your help <3

**Comment by putera** [comment]
Try use computed property

Component 
```php
public function getPaymentMethodProperty()
{
      return PaymentMethod::all();
}
```

Blade
```blade
<x-select
    label="Select a payment method"
    placeholder="Select an option"
    :options="$this->paymentMethod"
    wire:model.defer="payment_method"
    option-label="name"
    option-value="id"
/>
```

**Comment by ph7jack** [comment, marked as answer]
It never will work because is not a valid blade attribute. 
You can use `:options="$methods"` instead

> I have tried this way but it doesn't work
> 
```
<x-select
    label="Select a payment method"
    placeholder="Select an option"
    :options="$methods"
    wire:model.defer="payment_method"
/>
```


Like the @putera says, you can use a computed property too.

My recommendation is:
Is it a large options entry? Like 30+?
If so, use the async search in the select https://livewire-wireui.com/docs/select#async-search

Is it a short options entry? 
If so, use this

LivewireComponent.php
```
public array $methods = [];

public function mount() 
{
    // use this way to prevent the livewire refreshing these values in each request. Put all as an array
    $this->methods = PaymentMethod::query()->select('id', 'name')->get()->toArray();
 }
```

livewire-component.blade.php
```
<x-select
    label="Select a payment method"
    placeholder="Select an method"
    :options="$methods"
    wire:model.defer="payment_method"
    option-label="name"
    option-value="id"
/>
```

**Reply by KevinEscobarV** [reply]
**Thank you so much** 
they should put a small example like this in the documentation.
